.github/workflows/release.yaml

```diff
@@ -14,8 +12,35 @@ on:
     - cron: '0 0 * * *'
   workflow_dispatch:
 
+permissions:
+  pull-requests: read
+
+env:
+  BUILD_LABEL: "run release build"
+
 jobs:
+  check-pr-labels:
+    name: check PR labels
+    runs-on: ubuntu-latest
+    outputs:
+      run_release_workflow: ${{ steps.gen_output.outputs.run_release_workflow }}
+    steps:
+      - id: gen_output
+        run: |
+          if [ "${{ github.event_name }}" = 'pull_request' ] ; then
+            run_it=$(if gh api --jq '.labels.[].name' /repos/${{ github.repository }}/pulls/${{ github.event.number }} | grep --quiet '^${{ env.BUILD_LABEL }}$' ; then printf "%s" "yes" ; else printf "%s" "no" ; fi)
+            echo "${run_it}"
+            echo run_release_workflow="${run_it}" >> "$GITHUB_OUTPUT"
+          else
+            echo run_release_workflow="yes" >> "$GITHUB_OUTPUT"
+          fi
+        shell: bash
+        env:
+          GH_TOKEN: ${{ github.token }}
+
   release-workflow:
+    needs: ["check-pr-labels"]
+    if: ${{ needs.check-pr-labels.outputs.run_release_workflow == 'yes' }}
```

Comment:
Thanks Julian, It's good to merge with this solution in place, but I wonder why a solution like this didn't work out.  https://stackoverflow.com/questions/62325286/run-github-actions-when-pull-requests-have-a-specific-label

Reply:
Oh well, that surely is simpler

Reply:
Oh... I just realized this isn't what we want. This would only run the action *once* when the PR is labeled and then never again. I find that kinda counter-intuitive. If there's a label => do stuff, if there is none => don't do stuff.